platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblems.java

```diff
@@ -62,6 +64,11 @@ public ProblemReporter getReporter() {
         return createReporter();
     }
 
+    @Override
+    public DeprecationReporter getDeprecationReporter() {
+        return new DefaultDeprecationReporter(createReporter());
```

Comment:
:question: Can we have single, shared `InternalProblemReporter` field?

Reply:
Good idea! Actually, with a second pass, I could greatly simplify the class.

Reply:
I don't want to resolve this without a second pass, as the change is non-trivial.